lib/fs.js

```diff
@@ -591,6 +605,72 @@ function writeSync(fd, buffer, offset, length, position) {
   return result;
 }
 
+// usage:
+// fs.writev(fd, buffer, [chunkLength, [offset, [length, [position]]]],
+//    callback);
+function writev(fd, buffer, chunkLength, offset, length, position, callback) {
+  function wrapper(err, written) {
+    // Retain a reference to chunks so that they can't be GC'ed too soon.
+    callback(err, written || 0, buffer);
+  }
+
+  validateUint32(fd, 'fd');
+
+  const req = new FSReqCallback();
+  req.oncomplete = wrapper;
+
+  callback = maybeCallback(callback || position ||
+     length || offset || chunkLength);
+
+  if (typeof position !== 'number')
+    position = null;
+
+  if (isBuffersArray(buffer)) {
+    return binding.writeBuffers(fd, buffer, position, req);
+  }
+
+  if (typeof chunkLength !== 'number')
+    chunkLength = 1024;
+  if (typeof offset !== 'number')
+    offset = 0;
+  if (typeof position !== 'number')
+    position = null;
+
+  if (isArrayBufferView(buffer) || isTypedArray(buffer)) {
+    const chunks = [];
+    if (typeof length !== 'number')
+      length = buffer.byteLength - offset;
+
+    validateOffsetLengthWrite(offset, length, buffer.byteLength);
+
+    buffer = buffer.slice(offset, offset + length);
+    for (var i = 0; i < buffer.byteLength; i += chunkLength) {
+      chunks.push(buffer.slice(i, i + chunkLength));
+    }
```

Comment:
I think @mscdex already mentioned this in passing but I don't understand why you're chopping up the input like that. 1024 is also on the low (read: inefficient) side.

Reply:
- I am chopping it like that to convert it to an array as `writeBuffers` accepts an array of buffers.
- What do you think is a good default value for chunkLength?

Reply:
Then why don't you pass in `[buffer]`, i.e., an array with just the original buffer? Optionally cloned-by-slicing if you're afraid of modifications while the write is still in-flight, but I don't think we do that anywhere else either.

If you feel a compelling case can be made that slicing is necessary, then I suggest going with 64k or more.